asdf/extension.py

```diff
@@ -113,8 +114,12 @@ def __init__(self, extensions):
         for extension in extensions:
             if not isinstance(extension, AsdfExtension):
                 raise TypeError(
-                    "Extension must implement asdftypes.AsdfExtension "
+                    "Extension must implement the asdf.extension.AsdfExtension "
                     "interface")
+            elif not issubclass(type(extension), AsdfExtension):
+                warnings.warn("Extensions not inheriting from asdf.extension.AsdfExtension"
```

Comment:
I think it would be helpful to have the name of the extension in the warning message if possible.